Cargo.toml

```
@@ -22,12 +22,20 @@ features = ["serde-serialize"]
 test = false
 
 [features]
-default = ["spans", "std"]
+default = ["spans", "std", "diagnostic"]
 enable-interning = ["std"]
 serde-serialize = ["serde", "serde_json", "std"]
 spans = ["wasm-bindgen-macro/spans"]
 std = ["wasm-bindgen-macro/std", "once_cell/std"]
 
+# Enables the use of Rust's `diagnostic` macro for several traits to provide
+# better error messages when those traits are not implemented.
+#
+# Diagnostic messages are only available since Rust 1.78. Enabling this feature
+# for older compilers will NOT improve error message and will NOT result in a
+# compilation error.
+diagnostic = ["rustversion"]
```

```suggestion
msrv = ["rustversion"]
```
We should just say here that this enables features that require a higher MSRV. The part where you clarify that this doesn't actually make compilation fail is great. Maybe use the word "compile-time detection" so users understand that they are opting into something here that will increase compilation time.

What this feature will actually do, until now only improve diagnostics, should be documented somewhere else and here we just leave a link to there.

---

And where should I put the diagnostics docs?

---

I have no idea! Ideally we have a place where we document all crate features, but I don't think we do!

I think the most appropriate place is top-level Rustdoc, so lets just start by at least documenting `msrv`.